src/ooxml/java/org/apache/poi/xssf/extractor/XSSFImportFromXML.java

```diff
@@ -160,6 +172,57 @@ public void importFromXML(String xmlInputString) throws SAXException, XPathExpre
             }
         }
     }
+    
+    /**
+     * Resize the table by setting a new row count.
+     * 
+     * If the new row count is less than the current row count, superfluous rows will be cleared.
+     * If the new row count is greater than the current row count, cells below the table will be overwritten by the table.
+     *
+     * @param table the table to modify
+     * @param newRowCount new row count for the table
+     */
+    private void resizeTable(XSSFTable table, int newRowCount) {
+        
+        table.updateReferences();
+        int rowCount = table.getRowCount();
+        if(rowCount == newRowCount) {
+            return;
+        }
+        
+        // calculate new area
+        SpreadsheetVersion spreadsheetVersion = table.getXSSFSheet().getWorkbook().getSpreadsheetVersion();
+        CellReference newEndCellReference = new CellReference(table.getStartCellReference().getRow() + newRowCount, table.getEndCellReference().getCol());
+        AreaReference newTableAreaReference = new AreaReference(table.getStartCellReference(), newEndCellReference, spreadsheetVersion);
+        
+        // clear cells
+        CellReference clearAreaStartCell;
+        CellReference clearAreaEndCell;
+        if(newRowCount < rowCount) {
+            // clear all table cells that are outside of the new area
+            clearAreaStartCell = new CellReference(newTableAreaReference.getLastCell().getRow() + 1, newTableAreaReference.getFirstCell().getCol());
+            clearAreaEndCell = table.getEndCellReference();
+        } else {
+            // clear all cells below the table that are inside the new area
+            clearAreaStartCell = new CellReference(table.getEndCellReference().getRow() + 1, newTableAreaReference.getFirstCell().getCol());
+            clearAreaEndCell = newEndCellReference;
+        }
+        AreaReference areaToClear = new AreaReference(clearAreaStartCell, clearAreaEndCell, spreadsheetVersion);
+        for (CellReference cellRef : areaToClear.getAllReferencedCells()) {
+            XSSFRow row = table.getXSSFSheet().getRow(cellRef.getRow());
+            if (row != null) {
+                XSSFCell cell = row.getCell(cellRef.getCol());
+                if(cell != null) {
+                    cell.setCellType(CellType.BLANK);
+                    cell.setCellStyle(null);
+                }
+            }
+        }
+        
+        // update table area
+        table.getCTTable().setRef(newTableAreaReference.formatAsString());
```

Comment:
We regard accessing the CT* generated classes as something we want to discourage.
Could you change the the XSSFTable class to expose a setRef method?

Reply:
I just noticed `setRef` is already exposed as `XSSFTable.setCellReferences`, so I'll use that.

I also had to use CTTable and CTTableColumn near line 143 though, to get the correct column index. Now I saw that I could have used `XSSFTable.findColumnIndex(String columnHeader)` instead of calculating the column index myself, but it seems I would still have to use `CTTableColumn` to get the column header name. 

I didn't want to make any API changes before, but I guess in this case it would be useful if `XSSFXmlColumnPr` exposes the column header name and maybe also provides its column index. What do you think?

Reply:
If you could add any APIs that you need, that would be great. We want to make the generated CT* classes only internal implementation details (allowing us to move away from xmlbeans code generation eventually).

Reply:
There is an ongoing effort to replace calls to the underlying `CT*` classes, so that these calls be limited to the class directly wrapping the XML object. 

Every attempt in that direction is welcome.